### DISCUSSION TITLE
One node can't schedule - MountVolume.MountDevice failed - driver name driver.longhorn.io not found in the list of registered CSI drivers

### DISCUSSION THREAD

**Discussion by vinid223** [longhorn/longhorn#7232] [Q&A]
Hello,

I've had a power cut and had to restart all my nodes. They all restarted correctly, but as soon as I enable a deployment, when they are scheduled to one specific node, they can't mount a volume.

All other nodes works, but that one keep giving me an error.

> MountVolume.MountDevice failed for volume "pvc-b25f0503-ccc1-45cb-84aa-d9decf8e2582" : kubernetes.io/csi: attacher.MountDevice failed to create newCsiDriverClient: driver name driver.longhorn.io not found in the list of registered CSI drivers

> Unable to attach or mount volumes: unmounted volumes=[deployment-persistent-storage], unattached volumes=[deployment-persistent-storage kube-api-access-pw2p4]: timed out waiting for the condition

I can cordon the node so my deployments are working and can mount the volumes on other nodes.

Everything looks fine inside longhorn ui.

Can anybody help me?

**Comment by mikevader** [comment]
I experience the same issue after the upgrade from K3S Version 1.27.8+k3s2 to 1.28.4+k3s2

The strange part is that some pods started up and seem to work fine even with this error where as others now failing to start up.

**Reply by mantissahz** [reply]
Hi @mikevader,
Could you check the status of longhorn-csi-plugin pods? Are they working normally or any abnormal logs in the pods?

> The strange part is that some pods started up and seem to work fine even with this error where as others now failing to start up.

Do the pods that work fine have PVCs with longhorn volumes?

**Comment by PatrickHuetter** [comment]
I have exactly the same problem since a few hours.

**Comment by thedaringdev** [comment]
Just encountered this myself. Upgrading longhorn to 1.6 as well as upgrading the engine on all of my pvc seems to have fixed the problem.

**Comment by serhiynovos** [comment]
I have the same behavior. Usually happens after RKE2 version update or I have the same when I had 2 nodes powered off. Beside this node which is generating it produces next logs.
<img width="1014" alt="Знімок екрана 2024-06-27 о 16 50 06" src="https://github.com/longhorn/longhorn/assets/9201885/a2b88d70-bd80-4ef3-bbb2-c456e7a9888a">

**Reply by DamiaSan** [reply]
What version of Longhorn are you using?

**Reply by qww-ygg** [reply]
I also encountered this problem while restarting the system. Do you have any solutions?

k8s  v1.29.4
longhorn v1.6.1

Sep 09 15:51:15 risc-verse-k8sw-02 kubelet[3240779]: E0909 15:51:15.693595 3240779 remote_runtime.go:751] "ReopenContainerLog from runtime service failed" err="rpc error: code = Unknown desc = docker does not support reopening container log files" containerID="07d36dc7239aa7bd5b54ef9a115676a4f6cd571ec272544472f0c1bbfefeb272"
Sep 09 15:51:15 risc-verse-k8sw-02 kubelet[3240779]: E0909 15:51:15.705272 3240779 remote_runtime.go:751] "ReopenContainerLog from runtime service failed" err="rpc error: code = Unknown desc = docker does not support reopening container log files" containerID="6cb024d707e5b30413b43fc3da9662ea83a090c1a2fe4b01a864bd93a810b757"
Sep 09 15:51:16 risc-verse-k8sw-02 kubelet[3240779]: I0909 15:51:16.176919 3240779 reconciler_common.go:172] "operationExecutor.UnmountVolume started for volume \"pvc-10f4005a-397c-4765-9647-7573951d429e\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-10f4005a-397c-4765-9647-7573951d429e\") pod \"2ff54c20-4c3c-4fe2-967b-f122e8ac8fa7\" (UID: \"2ff54c20-4c3c-4fe2-967b-f122e8ac8fa7\") "
Sep 09 15:51:16 risc-verse-k8sw-02 kubelet[3240779]: E0909 15:51:16.177068 3240779 nestedpendingoperations.go:348] Operation for "{volumeName:kubernetes.io/csi/driver.longhorn.io^pvc-10f4005a-397c-4765-9647-7573951d429e podName:2ff54c20-4c3c-4fe2-967b-f122e8ac8fa7 nodeName:}" failed. No retries permitted until 2024-09-09 15:53:18.177036644 +0800 CST m=+1500916.668102124 (durationBeforeRetry 2m2s). Error: UnmountVolume.TearDown failed for volume "pvc-10f4005a-397c-4765-9647-7573951d429e" (UniqueName: "kubernetes.io/csi/driver.longhorn.io^pvc-10f4005a-397c-4765-9647-7573951d429e") pod "2ff54c20-4c3c-4fe2-967b-f122e8ac8fa7" (UID: "2ff54c20-4c3c-4fe2-967b-f122e8ac8fa7") : kubernetes.io/csi: Unmounter.TearDownAt failed to get CSI client: driver name driver.longhorn.io not found in the list of registered CSI drivers

**Comment by EugenMayer** [comment]
We encountered the same issue, with 3 different pods. We are running 1.30.6 rke2 with longhorn 1.7.2.

One can remove the pod and the thing will 'work again' - but it happened with 3 pods now, at different times and manually removing the pod cannot be a solution to this issue.

We did never encounter this issue before, this is as recent as the last 3 weeks, so it is:
- either related to longhorn 1.7.x
- related to use going from 1.29.x to 1.30 - so it might be a k8s side effect

never the less, this is an important issue that has been mentioned in #9424 and is revived by me in #9763
